pkg/resourcemonitor/podresourcesscanner.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourcemonitor
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	v1 "k8s.io/api/core/v1"
+	podresourcesapi "k8s.io/kubelet/pkg/apis/podresources/v1"
+)
+
+type PodResourcesScanner struct {
+	namespace         string
+	podResourceClient podresourcesapi.PodResourcesListerClient
+}
+
+func NewPodResourcesScanner(namespace string, podResourceClient podresourcesapi.PodResourcesListerClient) (ResourcesScanner, error) {
+	resourcemonitorInstance := &PodResourcesScanner{
+		namespace:         namespace,
+		podResourceClient: podResourceClient,
+	}
+	if resourcemonitorInstance.namespace != "" {
+		log.Printf("watching namespace %q", resourcemonitorInstance.namespace)
+	} else {
+		log.Printf("watching all namespaces")
+	}
+
+	return resourcemonitorInstance, nil
+}
+
+// isWatchable tells if the the given namespace should be watched.
+func (resMon *PodResourcesScanner) isWatchable(podNamespace string) bool {
+	if resMon.namespace == "" {
+		return true
+	}
+	// TODO:  add an explicit check for guaranteed pods
+	return resMon.namespace == podNamespace
+}
```

Comment:
I probably asked this already, but, what is the business with the "namespace-sensitivity" here?

Reply:
Please refer to the response [here](https://github.com/kubernetes-sigs/node-feature-discovery/pull/525/files#r668344868).